### DISCUSSION TITLE
How to update and use a variable everytime a FastAPI is hit

### DISCUSSION THREAD

**Discussion by parasjain06** [fastapi/fastapi#7313] [Questions] [answered]

**Comment by parasjain06** [comment]
I want to access a global variable when a fastAPI is hit, and at the end of API i want to update that global variable.

**Comment by Mause** [comment]
Have you looked into middlewares? https://fastapi.tiangolo.com/tutorial/middleware/

**Comment by parasjain06** [comment]
yes but does this maintain the state?
say i want to increment a counter everytime a request is hit ,
how can i do that?

lets say i want to update below dictionary for every request , and also maintain the state of the dictionary:- 
for ex:-
{"counter": 0, "yawnStatus": "False", "yawnCount": 0, "prevYawnStatus": "False"}

for each request, i want to update this counter and update other variables in dictionary so that in next request i can use them.

**Comment by Mause** [comment]
You could store it in the app.state attribute? 

So
```py
request.app.state.whatever = {"counter": 0}
```

**Comment by parasjain06** [comment]
![image](https://user-images.githubusercontent.com/63651642/94236346-b2b2e800-ff2a-11ea-9bd9-90e7de5ab76d.png)
 
I did this, but counter is not incrementing,
it is printing 1 everytime for each request, however should goes like:- 1,2,3,4,......

**Comment by Mause** [comment, marked as answer]
Try something like

```py

app.state = 0

def wellness(request: Request):
    request.app.state.counter += 1

```

**Comment by parasjain06** [comment]
![image](https://user-images.githubusercontent.com/63651642/94237113-f528f480-ff2b-11ea-9e19-e2137b25e1cf.png)
 
It worked , thanks a lot..!!!!! (y)

**Comment by Mause** [comment]
Feel free to close the issue :)

**Comment by tiangolo** [comment]
Thanks for the help @Mause! ☕ 

And thanks for coming back to close the issue @parasjain06 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
